drivers/dma/edma_mcux.c

```diff
@@ -0,0 +1,380 @@
+	edma_transfer_config_t edma_t_cfg;
+	edma_transfer_type_t type;
+	status_t status;
+
+	if ((channel + 1) > CONFIG_DMA_MCUX_NUMOF_DMA_CHANNELS) {
+		SYS_LOG_ERR("unsupported channel %d", channel);
+		return -EINVAL;
+	}
+
+	if (dma_cfg->channel_direction == MEMORY_TO_MEMORY) {
+		type = kEDMA_MemoryToMemory;
+		dma_cfg->dma_slot = (u8_t)(kDmaRequestMux0AlwaysOn63 & 0x3f);
+		SYS_LOG_DBG("direction: MEMORY_TO_MEMORY");
+	} else if (dma_cfg->channel_direction == MEMORY_TO_PERIPHERAL) {
+		type = kEDMA_MemoryToPeripheral;
+		SYS_LOG_DBG("direction: MEMORY_TO_PERIPHERAL");
+	} else if (dma_cfg->channel_direction == PERIPHERAL_TO_MEMORY) {
+		type = kEDMA_PeripheralToMemory;
+		SYS_LOG_DBG("direction: PERIPHERAL_TO_MEMORY");
+	} else {
+		return -EINVAL;
+	}
+
+	DMAMUX_SetSource(DMAMUX0, channel, dma_cfg->dma_slot);
+
+	/* TODO: support for the period trigger ? */
+	/*DMAMUX_EnablePeriodTrigger(DMAMUX0, channel);*/
+
+	DMAMUX_EnableChannel(DMAMUX0, channel);
+
+	EDMA_CreateHandle(&data[channel].handle, base, channel);
+	EDMA_SetCallback(&data[channel].handle, edma_mcux_callback, dev);
+	/* TODO: necessary ? */
+	/*EDMA_ResetChannel(base, channel); ? */
+
+	_install_tcd_memory(&data[channel].handle, channel);
+
+	data[channel].dma_callback = dma_cfg->dma_callback;
+	SYS_LOG_DBG("source_data_size %d", dma_cfg->source_data_size);
+	SYS_LOG_DBG("dest_data_size %d", dma_cfg->dest_data_size);
+	SYS_LOG_DBG("source_burst_length %d", dma_cfg->source_burst_length);
+	SYS_LOG_DBG("block_size %d", block_cfg->block_size);
+
+	EDMA_PrepareTransfer(&edma_t_cfg,
+			(void *)block_cfg->source_address,
+			(u32_t)dma_cfg->source_data_size,
+			(void *)block_cfg->dest_address,
+			(u32_t)dma_cfg->dest_data_size,
+			(u32_t)dma_cfg->source_burst_length,
+			(u32_t)block_cfg->block_size,
+			type);
+
+	status = EDMA_SubmitTransfer(&data[channel].handle, &edma_t_cfg);
+
+	if (status != kStatus_Success) {
+		SYS_LOG_ERR("Transfer could not submit");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int dma_mcux_start(struct device *dev, u32_t channel)
+{
+	struct dma_mcux_data *data = dev->driver_data;
+
+	if ((channel + 1) > CONFIG_DMA_MCUX_NUMOF_DMA_CHANNELS) {
+		SYS_LOG_ERR("unsupported channel %d", channel);
+		return -EINVAL;
+	}
+
+	EDMA_StartTransfer(&data[channel].handle);
+
+	return 0;
+}
+
+static int dma_mcux_stop(struct device *dev, u32_t channel)
+{
+	struct dma_mcux_data *data = dev->driver_data;
+
+	SYS_LOG_DBG("");
+
+	EDMA_StopTransfer(&data[channel].handle);
+
+	return 0;
+}
+
+static const struct dma_driver_api dma_mcux_driver_api = {
+	.config = dma_mcux_channel_config,
+	.start = dma_mcux_start,
+	.stop = dma_mcux_stop,
+};
+
+static int dma_mcux_configure(struct device *dev)
+{
+	const struct dma_mcux_config *config = dev->config->config_info;
+	struct dma_mcux_data *data = dev->driver_data;
+	DMA_Type *base = config->base;
+	edma_config_t edma_config;
+
+	SYS_LOG_DBG("");
+
+	data[0].dma_callback = NULL;
```

Comment:
Should not be there, forgotten to remove, as I see it is not necessary to initialize it with zeros.